### ISSUE TITLE
Support for NEEWER-GL1 PRO Color Temperature: 2900K~7000K

### ISSUE THREAD

**Issue by CV8R** [taburineagle/NeewerLite-Python#87]
Hi,

I have purchased a NEEWER GL1 PRO and logs show that cannot set to Color Temperature: 2900K.  I am reporting this as some feedback only because it appears that it sets the light to the lowest color temp anyway (compared with native app).

reports: `>> --temp (29) isn't between the bounds of 32 and 85, so falling back to closest start boundary of 32`

[https://neewer.com/products/neewer-gl1-pro-15-5-key-light-streaming-light-66602110](url)

Dimming Range: 0~100
Color Temperature: 2900K~7000K

This is a CCT only light.

Initially I was using the main branch and could only turn the light on and off, no dimming our color temp.  I then found this issue and post which got it working [https://github.com/taburineagle/NeewerLite-Python/issues/36#issuecomment-1079921234](url).

I am now using the WIP-HTTP-Server-Mods branch and same note in the logs.

Cheers and thanks!

**Comment by taburineagle** [comment]
Hi @CV8R - are you saying that you can set the color temperature and brightness, you just can't dial it down to 2900K?

**Comment by CV8R** [comment]
Correct, at least it reports that in the logs.  But visually, I think its working when compared to the native app.

**Comment by taburineagle** [comment]
Got you - I know what's happening then!  For some reason (I don't think I've quite gotten the entire way to changing the entire CLI/HTTP interface with the changes I've done over the past few months), the values of 3200K and 8500K are hard-coded into the tester for that parameter, so the program _sees_ 2800K as invalid, even though it isn't, specifically this line:

`testValid("temp", args.temp, 56, 32, 85),` - it tests to see if the value is between 32(00K) and 85(00)K, and returns false if it isn't... I need to change this in the next commit.

Also, one other question - how exactly does your light show up in NeewerLite-Python (in other words, is it listed as "GL1PRO", "GL1 Pro", "GL1 PRO", etc.)?  I have a section of pre-defined light parameters that _should_ recall the correct settings for that light, but apparently it's not pulling the right data as it's not listed in that list, hence the initial issue where the light would only turn on/off.

**Comment by CV8R** [comment]
The light is identified as NEEWER-GL1 PRO - `Successful link on light [NEEWER-GL1 PRO] MAC Address: `

**Comment by taburineagle** [comment]
Got you - OK, if you use the latest `WIP-HTTP-Server-Mods` commit, and delete the custom preference file, does that come up with the correct parameters for the light?  It's still going to give that `--temp (29)` error if you use a color temperature under 3200K until I change that in the next commit, but that version _should_ properly set the other parameters correctly (that it's a CCT-only light that uses a color temp range of 2900-7000K) as it does have CL1 in the master list of lights, and the CL1 Pro uses the same parameters as the CL1.

**Comment by CV8R** [comment]
> Got you - OK, if you use the latest `WIP-HTTP-Server-Mods` commit, and delete the custom preference file, does that come up with the correct parameters for the light? It's still going to give that `--temp (29)` error if you use a color temperature under 3200K until I change that in the next commit, but that version _should_ properly set the other parameters correctly (that it's a CCT-only light that uses a color temp range of 2900-7000K) as it does have CL1 in the master list of lights, and the CL1 Pro uses the same parameters as the CL1.)

I believe so yes, visually, when compared to the native app the color temp looks correct. When you push the next commit I will try it again to ensure the log is not reporting `--temp (29)` error.

**Comment by taburineagle** [comment]
OK, I've pushed a new commit to the `WIP-HTTP-Server-Mods` branch - it extends the range of CCT temperatures settable by the CLI/HTTP interface to 2500K-10000K... this should take care of the issue you were having before @CV8R.

**Comment by CV8R** [comment]
Apologies for the delay.  I have been traveling.

The color temp errors are now gone in the log and it the temp is changed on the light.  I note though, that if I add &on to turn it on the color temp and brightness are ignored.  Is there a way to go from off to on and set the brightness and temp?

**Comment by taburineagle** [comment]
Hi @CV8R - ha, well, my delay is a lot longer 😄  Sorry about that!  Yeah, at the moment, &on and &off will only turn the light on or off, and not change any settings.  This is something I want to take a look at for later versions though!
